### PR TITLE
Feature/spring boot 2.x

### PR DESCRIPTION
**Pull request by jwomeara**

### CODE REVIEW

**Comment by jwomeara** [comment]
The following PRs are all part of the feature/spring-boot-2.x effort and should be approved/merged together.

https://github.com/NationalSecurityAgency/datawave-accumulo-service/pull/8
https://github.com/NationalSecurityAgency/datawave-audit-service/pull/11
https://github.com/NationalSecurityAgency/datawave-authorization-service/pull/17
https://github.com/NationalSecurityAgency/datawave-config-service/pull/2
https://github.com/NationalSecurityAgency/datawave-dictionary-service/pull/17
https://github.com/NationalSecurityAgency/datawave-hazelcast-service/pull/3
https://github.com/NationalSecurityAgency/datawave-query-metric-service/pull/9
https://github.com/NationalSecurityAgency/datawave-spring-boot-starter-audit/pull/2
https://github.com/NationalSecurityAgency/datawave-spring-boot-starter-cache/pull/1
https://github.com/NationalSecurityAgency/datawave-spring-boot-starter/pull/14
https://github.com/NationalSecurityAgency/datawave-spring-boot-starter-query-metric/pull/3
https://github.com/NationalSecurityAgency/datawave-in-memory-accumulo/pull/11
https://github.com/NationalSecurityAgency/datawave-accumulo-utils/pull/6
https://github.com/NationalSecurityAgency/datawave-base-rest-responses/pull/3
https://github.com/NationalSecurityAgency/datawave-common-utils/pull/2
https://github.com/NationalSecurityAgency/datawave-metadata-utils/pull/18
https://github.com/NationalSecurityAgency/datawave-metrics-reporter/pull/4
https://github.com/NationalSecurityAgency/datawave-type-utils/pull/18
https://github.com/NationalSecurityAgency/datawave-spring-boot-starter-metadata/pull/2
https://github.com/NationalSecurityAgency/datawave/pull/1876

File: pom.xml [hunk 1]
```diff
@@ -147,63 +160,35 @@
...
+            <dependency>
+                <groupId>org.springdoc</groupId>
+                <artifactId>springdoc-openapi-ui</artifactId>
+                <version>${version.springdoc}</version>
             </dependency>
-            -->
         </dependencies>
     </dependencyManagement>
     <dependencies>
+        <!-- TODO: JWO: this breaks serialization of wrapped elements in MapReduceJobDescription.java and QueryLogicDescription.java -->
```
Comment (apmoriarty) [inline pom.xml:187]: still an issue?

File: src/test/java/datawave/microservice/config/web/DatawaveServerPropertiesTest.java [hunk 2]
```diff
@@ -83,142 +78,136 @@ public void testWithSslDisabledAndUnsetProperties() {
...
     @Test
     public void testWithSslEnabledAndMissingKeyStore() {
-        expectedException.expect(BeanCreationException.class);
-        expectedException.expectCause(new CauseMessageMatcher(BindValidationException.class, "Field error in object 'server' on field 'outboundSsl.keyStore'"));
-        // @formatter:off
-        TestPropertyValues.of(
+        BeanCreationException expectedException = assertThrows(BeanCreationException.class, () -> {
+            // @formatter:off
+            TestPropertyValues.of(
 //                "server.outbound-ssl.keyStore=testKeyStore",
```
Comment (apmoriarty) [inline src/test/java/datawave/microservice/config/web/DatawaveServerPropertiesTest.java:84]: can this commented line get deleted?

**Review by apmoriarty — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (jwomeara) [reply pom.xml:187]: This was something I removed once.  Then someone added it back, so I commented it out and left the comment to hopefully prevent them from adding it back again...  haha.  I'll remove it.

**Pull request closed without merging**